jetty-documentation/src/main/asciidoc/administration/sessions/session-configuration-file-system.adoc

```diff
@@ -0,0 +1,60 @@
+//  ========================================================================
+//  All rights reserved. This program and the accompanying materials
+//  are made available under the terms of the Eclipse Public License v1.0
+//  and Apache License v2.0 which accompanies this distribution.
+//
+//      The Eclipse Public License is available at
+//      http://www.eclipse.org/legal/epl-v10.html
+//
+//      The Apache License v2.0 is available at
+//      http://www.opensource.org/licenses/apache2.0.php
+//
+//  You may elect to redistribute this code under either of these licenses.
+//  ========================================================================
+
+[[configuring-sessions-file-system]]
+
+=== Configuring Session Management using the File System
+
+When using the Jetty distribution, you will first need to enable the `session-store-file` link:#startup-modules[module] for your link:#startup-base-and-home[Jetty base] using the `--add-to-start` argument on the command line.
+
+[source, screen, subs="{sub-order}"]
+----
+$ java -jar ../start.jar --add-to-start=session-store-file
+INFO : server          initialised (transitively) in ${jetty.base}/start.d/server.ini
+INFO : sessions        initialised (transitively) in ${jetty.base}/start.d/sessions.ini
+INFO : session-store-file initialised in ${jetty.base}/start.d/session-store-file.ini
+MKDIR: ${jetty.base}/sessions
+INFO : Base directory was modified
+----
+
+//TODO - Callout default Session file location - note it is configurable
+Doing this enables the File System Session module and any dependent modules or files needed for it to run on the server.
+The example above is running an fresh `{$jetty.base}` with nothing enabled.
+When the `--add-to-start` argument was added to the command line, it enabled the the `session-store-file` module as well as the `sessions` and `server` modules, which are required for the File System session management to operate.
+Additionally a `${jetty.base}/sessions` directory was created.
+By default Session files will be saved to this directory.
+
+In addition to adding these modules to the classpath of the server it also added several ini configuration files to the `start.d` directory of the `{$jetty.base}`.
+
+Opening the `start.d/session-store-file.ini` will show a list of all the configurable options for the JDBC module:
+
+[source, screen, subs="{sub-order}"]
+----
+# ---------------------------------------
+# Module: session-store-file
+# Enables session persistent storage in files.
+# ---------------------------------------
+--module=session-store-file
+
+jetty.session.storeDir=${jetty.base}/sessions
+#jetty.session.deleteUnrestorableFiles=false
+----
+
+jetty.session.storeDir::
+This defines the location for storage of Session files.
+jetty.session.deleteUnrestorableFiles::
+Boolean.
+If set to true, the server will delete files during the scavenging process for Sessions that have expired or otherwise unable to be restored.
```

Comment:
If set to true, unreadable files will be deleted: this is useful to prevent repeated logging of the same error when the scavenger periodically (re-) attempts to load the corrupted information for a session in order to expire it.